lib/logfmt/tagged_logger.rb

```diff
@@ -7,20 +7,9 @@ module Logfmt
   class TaggedLogger < ActiveSupport::Logger
     include ActiveSupport::TaggedLogging
 
-    def initialize(*args, formatter: nil, **kwargs)
+    def initialize(*args, formatter: KeyValueFormatter.new, **kwargs)
```

Comment:
IIUC, the change here (as compared to the original suggestion in the code comment) is that we're relying on [the underlying `::Logger#initialize` behavior](https://github.com/ruby/ruby/blob/234722ed9fce33d7ec8152920636af663d527912/lib/logger.rb#L579-L586), which accepts a `formatter:` kwarg to have already set up `self.logger` for us here. Yeah?

I suppose the only difference in behavior then is that if someone explicitly passed `formatter: nil`, for example, they'd get exactly that, and that's `::Logger`'s default for that argument too. So… I think they'd end up with a `::Logger::Formatter`, which would be wholly incompatible with the tagged logger. But then, why were you passing `nil`, I suppose?

I don't know, I guess I'm "six of one, half a dozen of the other" on this one (as my grandad used to say).

Reply:
I agree with that description. 

I’m coming around to the original comment’s suggestion because it is quite clear what it does with the formatter, whereas the kwarg slightly obscures it. I was too clever maybe

Reply:
Updating this PR per the above. But we're not going to merge it yet as it requires Rails 7.1+, which also means dropping older Rubies. We'll get there, eventually, but probably closer to the EoL of Rails 7.1